### ISSUE TITLE
Installation for Asus Zenfone 2 devices (use i686 instead of x86_64)

### ISSUE THREAD

**Issue by gauloics** [sdrausty/TermuxArch#128]
Installation for Asus Zenfone 2 devices (i686). Change or enter a new mirror 
`Server = https://mirror.archlinux32.org/$arch/$repo` during installation or after the installation is complete you can change it manually in arch **/etc/pacman.d/mirrorlist**.
If your screen appears like **[root18: 24arch] $** then while you can ignore the error(temporarily). Then open the **/etc/resolv.conf** file to make sure there are 
>**nameserver 8.8.8.8
nameserver 8.8.4.4**

if it is not there then write it into the file and save it. Then open the **etc/pacman.conf** file and change the **Architecture** value from **auto** to **i686**, change the **SigLevel** value from **Required DatabaseOptional** to **Never** (only temporarily), then save.
Then try run
`keys`
`pci`
maybe there some error, ignore it.

Then run 
`pacman -S ca-certificates-utils`
`pacman -Syu`
>[root15:19arch]$ pacman -S ca-certificates-utils
resolving dependencies...
looking for conflicting packages...                                                       
Package (1)                 Old Version  New Version   Net Change                         
core/ca-certificates-utils  20160507-1   20180821-1.0    0.08 MiB
Total Installed Size:  0.09 MiB
Net Upgrade Size:      0.08 MiB                                                           
:: Proceed with installation? [Y/n]

pres Y then enter...
if you see like
>:: Processing package changes...
(1/1) upgrading ca-certificates-utils [############################] 100%
:: Running post-transaction hooks...
(1/2) Arming ConditionNeedsUpdate...
(2/2) Rebuilding certificate stores...

it's mean installation complete
Then run
`pacman -Syu`
Maybe still have some warning and 1 error.
After full upgrade is complete, check the history during installation, if there is a warning stating that there is a new file created with **filename.pacnew** then use the filemanager to replace the old file with the newly created file.
>For example, a new file named **pacman.conf.pacnew**, then change the old file name **pacman.conf** to **pacman.conf.bak** (for backup if there is a problem can be returned immediately), then rename the new file **pacman.conf.pacnew** to **pacman.conf** .

Especially for the **pacman.conf** file, after you change the **pacman.conf.pacnew** file to **pacman.conf**, don't forget to change the value from **Architecture** from **auto** to **i686**, but don't need to change the SigLevel part.
For **mirrorlist** file, don't forget to uncomment one of the server list.
Also,,, Just ignore warning for this section, don't need to change file name:
>warning: /etc/group installed as /etc/group.pacnew
warning: /etc/gshadow installed as /etc/gshadow.pacnew 
warning: /etc/passwd installed as /etc/passwd.pacnew
warning: /etc/resolv.conf installed as /etc/resolv.conf.pacnew 
warning: /etc/shadow installed as /etc/shadow.pacnew

Update server gpg-agent by runing
`rm -rf arch/etc/pacman.d/gnupg`
`keys`

If you've error about locale like
>/usr/bin/locale-gen: line 33: sed: command not found
  .UTF-8/usr/bin/locale-gen: line 35: sed: command not found
.../usr/bin/locale-gen: line 38: sed: command not found

run `pacman -Syu sed`, may there some other pakages installation. Then run `locale-gen`

Try one more time run `pacman -Syu`, if you can see
>[root18:24arch]$ pacman -Syu
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
:: Starting full system upgrade...
 there is nothing to do

**means you are successful and archlinux is in your hands**

**Comment by gauloics** [comment]
Still so far for one tap installation
1. The mirrorlist of the archlinux and archlinux32 repositories is different, so that installing the archlinux32 package will fail if you use the default mirrorlist.
2. If the mirrorlist has been replaced, the installation is blocked by an imperfect keyring, so it will still change the architecture value from auto to i686 and negate the SigLevel first.
3. After the second stage is still hindered because of the conflict of ca-certificates-utils: /etc/ssl/certs/ca-certificates.crt exists in the filesystem, it must delete the file first. And you have to install the package manually and individually before installing another package, because it's like a boarding pass for other packages.
4. Sed for locale-gen also doesn't buildin, it must be installed manually

**Comment by gauloics** [comment]
If there is time, actually I also want to contribute, while learning coding, but I still do not understand about github repository, whether when I use git clone then any code changes directly affect the source code?
I tried reading the help on the github website, but it was too much and I didn't know where to start if I wanted to learn to be specific about this case. And believe me I'm still too new to coding. Just for learning for myself, what if I want to try tweaking the code so I can run it on the same architecture as my device but it doesn't directly affect the parent repository before it is truly perfect?

**Comment by sdrausty** [comment]
> Still so far for one tap 

installation and configuration.

Can you run `setupTermuxArch.sh manual altdir` to check whether the system is being detected correctly.  Edit \_I686_() in setupTermuxArchConfigs.sh like so:
```
_I686_() {
        echo found i686
        exit 86
```
And post your output.  The run should exit if detection went smoothly.  If we take it one step at a time, this goal is achievable for this architecture.

> I tried reading the help on the github website

Use keywords in a search engine to help.  There is a lot of information in the Internet about GitHub.  Also, are you familiar with [Instructional GitHub and Git Foundations Videos Can Be Downloaded for Viewing at Your Leisure](https://sdrausty.github.io/pages/ggf.html)?

**Comment by sdrausty** [comment]
> Still so far for one tap 

installation and configuration.

Can you run `setupTermuxArch.sh manual altdir` to check whether the system is being detected correctly.  Edit \_I686_() in setupTermuxArchConfigs.sh like so:
```
_I686_() {
        echo found i686
        exit 86
```
And post your output.  The run should exit with Code 86 if detection went smoothly.  If we take it one step at a time @gauloics, this goal is achievable for this architecture.

> I tried reading the help on the github website

Use keywords in a search engine to help.  There many tutorials in the Internet about GitHub.  Also, are you familiar with [Instructional GitHub and Git Foundations Videos Can Be Downloaded for Viewing at Your Leisure](https://sdrausty.github.io/pages/ggf.html)?  This feature is abbreviated to the command `yt` in TermuxArch Arch Linux.  

>  Enter the following to dowload this video series into your favorite folder:

$ yt https://m.youtube.com/playlist?list=PL0lo9MOBetEHhfG9vJzVCTiDYcbhAiEqL

> This way you will be able to view Git and GitHub Foundations from your favorite folder at your leisure.

**Comment by gauloics** [comment]
**add note**
Not support blackarch because don't have i686 repositories

**Comment by gauloics** [comment]
>Can you run setupTermuxArch.sh manual altdir to check whether the system is being detected correctly. Edit _I686_() in setupTermuxArchConfigs.sh like so:

_I686_() {
        echo found i686
        exit 86
And post your output. The run should exit with Code 86 if detection went smoothly. If we take it one step at a time @gauloics, this goal is achievable for this architecture. One tap installation and configuration is amazing to observe on arm architectures.

I followed this and the script still downloads this file, 
>🕛> 🕓 Downloading archlinux-bootstrap-2017.03.01-i686.tar.gz from http: //archive.archlinux.org… This may take a long time pending connection.

**Comment by sdrausty** [comment]
> [Detected x86_64 Android operating system.](https://github.com/sdrausty/TermuxArch/issues/47#issuecomment-427222438)

@gauloics this might sound redundant; for this device the line should read, "Detected x86 (i686) Android operating system."  Do I understand you correctly on this issue?  Can you run `termux-info`, and share the results also?

**Comment by sdrausty** [comment]
@gauloics this commit https://github.com/sdrausty/TermuxArch/commit/9d622785a9122d999213dee2431cdbede7eaf7a1#diff-2ddec689150ab81d43c07b5aae0ccb1dL129 changes the output message in  \_PRINT_DETECTED_SYSTEM_().  This function used to get system information from uname.  Now it gets this information from getprop.  How does it display on the ASUS?

**Comment by sdrausty** [comment]
@gauloics this https://github.com/sdrausty/TermuxArch/commit/f4b2685fc6499e3957e44ec959df7a25eec1dd7d rewrites the flavor option for Alpine and Arch.  Can you check whether system detection is correct for both flavors?

**Comment by gauloics** [comment]
>$ termux-info
Updatable packages:
All packages up to date
System information:
Linux localhost 3.10.20-x86_64_moor-265462-gfd33dd4 #1 SMP PREEMPT Fri Sep 30 15:31:23 CST 2016 x86_64 Android
Termux-packages arch:
i686
Android version:
5.0
Device manufacturer:
asus
Device model:
ASUS_Z00A

**Comment by gauloics** [comment]
>@gauloics this might sound redundant; for this device the line should read, "Detected x86 (i686) Android operating system." Do I understand you correctly on this issue? Can you run termux-info, and share the results also?

That's why I tried today to change Pacman.conf on the architecture part of i686 to auto, where it automatically detects x86_64, with the same steps when the initial setup is as above, but fails, damage occurs 😭😭😭.

**Comment by gauloics** [comment]
my new installation detects x86, and not x86_64.
>ＴｅｒｍｕｘＡｒｃｈ v1.6.4.id6300

**Comment by gauloics** [comment]
I tried switching to x86_64 because when I use gradle I always have problems which if I look for information from google is an architectural problem. Strangely, the problem arises from the 64 architecture and must install something lib32 bla bla bla

**Comment by gauloics** [comment]
in your opinion, should my device use i686 (archlinux32) or x86_64? What do you think if I want to try the termuxArch installation directly to download x86_64 as well as Alpine, which mirror can be directly changed?

**Comment by sdrausty** [comment]
> should my device use i686 (archlinux32) or x86_64? 
>> Termux-packages arch:
i686

i686=x86

> termuxArch installation directly to download x86_64

Change these lines to read the detect architecture:
```
declare CPUABI5="armeabi"       ## Used for development.                                  
declare CPUABI7="armeabi-v7a"   ## Used for development.                                  
declare CPUABI8="arm64-v8a"     ## Used for development.                                  
declare CPUABIX86="x86"         ## Used for development.                                  
declare CPUABIX86_64="x86_64"   ## Used for development.
```
To install x86_64 on x86 use:
```
declare CPUABIX86="changed"  ## Used for development.                                  
declare CPUABIX86_64="x86"   ## Used for development.
```
This will cause TermuxArch to install x86_64 in x86.  This will only work in the repository and TermuxArchBloom.  Have you read up on the 32 64 bit mixed architecture issues at Termux?  There are a few in temux-app and termux-packages.

**Comment by gauloics** [comment]
>Have you read up on the 32 64 bit mixed architecture issues at Termux? There are a few in temux-app and termux-packages.

Never before, are there references? And this architectural problem is quite confusing, which I once asked. https://github.com/sdrausty/TermuxArch/issues/121#issuecomment-422718838

**Comment by sdrausty** [comment]
This https://github.com/sdrausty/TermuxArch/commit/fb9e6df0d4228671949e14ddbcee2f4880a8ee3b commit fixes md5sum error for x86 and x86_68 with Arch Linux.

**Comment by gauloics** [comment]
got
>proot warning: can't chdir("/data/sdext2/data/com.termux/files/home/arch/./.") in the guest rootfs: No such file or directory
proot info: default working directory is now "/"
proot error: '/data/sdext2/data/com.termux/files/home/arch/usr/bin/env' is a 64-bit program whereas this version of proot handles 32-bit programs only
proot info: Get a 64-bit version that supports 32-bit binaries here: http://static.proot.me/proot-x86_64
proot info: vpid 1: terminated with signal 11

is this related to the script? if yes where should we change?

*As information, the bloom script has made several files outside the TermuxArchBloom folder, is this normal? Because I remember this file did not appear when I made a bloom before*

**Comment by sdrausty** [comment]
> is this related to the script? about 

@gauloics I don't know.  I would have had a better chance of providing an answer if you had given input also.
>  I remember this file did not appear when I made a bloom before

Ditto.  The most recent development https://github.com/sdrausty/gensTermuxArch/commit/928e57fc2045a9028e018a160323aa957854f972 version is attempting to address this issue once again.  You are welcome to test along.  

>> : Get a 64-bit version that supports 32-bit binaries here

Have you researched this topic upstream at https://github.com/termux?

**Comment by sdrausty** [comment]
@gauloics are you familiar with https://github.com/termux/x11-packages/tree/master/packages/qemu-system-x86_64?  This repository https://github.com/termux/x11-packages was published last month.

**Comment by sdrausty** [comment]
This is the results for gen:

```
$ ./setupTermuxArch.sh b

Setting mode to bloom.
total 12
drwx------    2 10106    10106         4096 Oct  8 15:25 .
drwx------   11 10106    10106         8192 Oct  8 15:25 ..

 🕛 > 🕛 ＴｅｒｍｕｘＡｒｃｈ gen.v1.6.4.id542149868067 bloom option.  Run bash setupTermuxArch.sh help for additional information.  Ensure background data is not restricted.  Check the wireless connection if you do not see one o'clock 🕐 below.  Checking prerequisites…

Found download tool: wget; Continuing…

Using wget to manage downloads.

 🕛 > 🕧 Prerequisites: OK  Downloading ＴｅｒｍｕｘＡｒｃｈ…

setupTermuxArch.sha512 100%[==========================>]     153  --.-KB/s    in 0s
setupTermuxArch.tar.gz 100%[==========================>]  38.58K   229KB/s    in 0.2s

 🕛 > 🕐 ＴｅｒｍｕｘＡｒｃｈ download: OK

 🕛 > 🕜 ＴｅｒｍｕｘＡｒｃｈ gen.v1.6.4.id542149868067 integrity: OK


TermuxArch: DIRECTORY WARNING!  /data/data/com.termux/files/home/TermuxArchBloom/ directory detected.  setupTermuxArch.sh bloom will continue.

Refresh /data/data/com.termux/files/home/TermuxArchBloom? [Y|n]
Uninstalling /data/data/com.termux/files/home/TermuxArchBloom…
Uninstalling /data/data/com.termux/files/home/TermuxArchBloom done.

ls: /data/data/com.termux/files/usr/tmp/setupTermuxArch6074/*: No such file or directory

 ＴｅｒｍｕｘＡｒｃｈ WARNING:  Generated script signal 1 near or at line number 54 by ``!

setupTermuxArch.sh b gen.v1.6.4.id542149868067: [Exit Signal 201] DONE 🏁

$ l TermuxArchBloom/
total 8
drwx------    2 10106    10106         4096 Oct  8 15:25 .
drwx------   37 10106    10106         4096 Oct  8 15:25 ..
$
```
```
$ l setupTermuxArch6074/
total 272
drwx------    2 10106    10106         4096 Oct  8 15:25 .
drwx------   11 10106    10106         8192 Oct  8 15:25 ..
-rw-r--r--    1 10106    10106        11810 Oct  5 19:05 LICENSE
-rw-r--r--    1 10106    10106         1103 Oct  5 19:05 README.md
-rw-r--r--    1 10106    10106        32833 Oct  6 08:57 archlinuxconfig.sh
-rw-r--r--    1 10106    10106        22094 Oct  8 14:45 espritfunctions.sh
-rw-r--r--    1 10106    10106         3814 Oct  8 07:16 getimagefunctions.sh
-rw-r--r--    1 10106    10106         6374 Oct  6 09:47 knownconfigurations.sh
-rw-r--r--    1 10106    10106        14882 Oct  6 06:39 maintenanceroutines.sh
-rw-r--r--    1 10106    10106        14812 Oct  8 07:13 necessaryfunctions.sh
-rw-r--r--    1 10106    10106        17951 Oct  8 14:45 printoutstatements.sh
-rwxr--r--    1 10106    10106        31665 Oct  8 14:45 setupTermuxArch.sh
-rw-------    1 10106    10106          153 Oct  8 15:25 setupTermuxArch.sha512
-rw-------    1 10106    10106        39510 Oct  8 15:25 setupTermuxArch.tar.gz
-rwx------    1 10106    10106        31665 Oct  8 15:25 setupTermuxArch.tmp
-rw-r--r--    1 10106    10106         1208 Oct  8 14:45 termuxarchchecksum.sha512
$ pwd
/data/data/com.termux/files/usr/tmp
```

I don't understand why this error happens.  Do you?

**Comment by sdrausty** [comment]
>> Have you read up on the 32 64 bit mixed architecture issues at Termux? There are a few in temux-app and termux-packages.

> Never before, are there references? 

https://github.com/termux/termux-app/issues?q=is%3Aissue+32+64+bit+architecture

https://github.com/termux/termux-app/issues?q=is%3Aissue+32+64+bit+architecture

**Comment by sdrausty** [comment]
This https://github.com/sdrausty/TermuxArch/commit/5760a4cc84c5b0ca398a01170a2513d765cba5b3 commit resolves:
> the bloom script has made several files outside the TermuxArchBloom folder

**Comment by sdrausty** [comment]
> I think attempts at problems related to the experiment forced installation of the x86_64 architecture to be better open on another issue.

This sentence is confusing.  Did either x86 or x86_64 work correctly on the ASUS?  If this is ASUS specific, TermuxArch can be set to install one instead of the other.

**Comment by gauloics** [comment]
>I think attempts at problems related to the experiment forced installation of the x86_64 architecture to be better open on another issue.

*I mean the x86_64 issue is separated from this issue, and makes another issue, and the issue with this title let it keep talking about i686.*

This is back talking about i686, I experienced the same problem with the issue that was opened recently. Please check your commit regarding the proot. Even though I have dumped all bloom files, and re-downloaded the script without changing it at all.

**1. Proot error**
> 🕛 > 🕘 Cleaning up installation files: DONE
 🕛 > 🕤 Arch Linux in Termux PRoot is installed.  Configuring and updating Arch Linux 📲
proot warning: can't chdir("/data/sdext2/data/com.termux/files/home/arch/./.") in the guest rootfs: No such file or directory
proot info: default working directory is now "/"
To generate locales in a preferred language use Settings > Language & Keyboard > Language in Android; Then run setupTermuxArch.sh r for a quick system refresh; For full system refresh use setupTermuxArch.sh refresh.

**2. Still about sed**
> ==> Generating locales...
/usr/bin/locale-gen: line 33: sed: command not found
  .UTF-8/usr/bin/locale-gen: line 35: sed: command not found
.../usr/bin/locale-gen: line 38: sed: command not found
:: Removing redundant packages for Termux PRoot installation…
==> Running TermuxArch keys archlinux-keyring archlinux32-keyring-transition ca-certificates-utils v1.6.5.id2161…

**3. Archlinux32 keyring**
> ==> Running pacman -Sy archlinux-keyring archlinux32-keyring-transition ca-certificates-utils --noconfirm --color=always…
:: Synchronizing package databases...
 core                     135.4 KiB   330K/s 00:00 [###################] 100%
 extra                   1638.4 KiB   329K/s 00:05 [###################] 100%
 community                  4.6 MiB   486K/s 00:10 [###################] 100%
error: target not found: archlinux32-keyring-transition

**4. During the installation process of the full system upgrade, part of the pci, Synchronizing package databases...**
> libidn2-2.0.5-1-x86_64   110.0 KiB  1000K/s 00:00 [###################] 100%
 libpsl-0.20.2-1-x86_64    71.9 KiB  1027K/s 00:00 [###################] 100%
 libnghttp2-1.33.0-1...    86.4 KiB   864K/s 00:00 [###################] 100%
kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
 curl-7.61.1-3-x86_64     939.1 KiB  1105K/s 00:01 [###################] 100%
 npth-1.6-1-x86_64         13.1 KiB  1310K/s 00:00 [###################] 100%
 libksba-1.3.5-1-x86_64   116.0 KiB  86.6K/s 00:01 [###################] 100%

**5. The last of the processes that are in the description number 4**
> arch-install-script...     8.9 KiB   445K/s 00:00 [###################] 100%
(90/90) checking keys in keyring                   [###################] 100%
downloading required keys...
kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
error: key "EEEEE2EEEE2EEEEE" could not be looked up remotely
error: key "4A1AFC345EBE18F8" could not be looked up remotely
error: required key missing from keyring
error: failed to commit transaction (unexpected error)
Errors occurred, no packages were upgraded.
error: no operation specified (use -h for help)
 pci WARNING:  Generated script signal 1 near or at line number 77 by !
TermuxArch pci gzip sed v1.6.5.id2161: DONE 🏁
error: key "EEEEE2EEEE2EEEEE" could not be looked up remotely
error: key "4A1AFC345EBE18F8" could not be looked up remotely
error: required key missing from keyring
error: failed to commit transaction (unexpected error)
error: no operation specified (use -h for help)
 finishsetup.sh WARNING:  Generated script signal 201 near or at line number 54 by
==> Running TermuxArch pci  v1.6.5.id2161…

**6. Second Synchronizing package databases...**
> :: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
:: Starting full system upgrade...
:: Replace ca-certificates-cacert with core/ca-certificates? [Y/n]
resolving dependencies...
looking for conflicting packages...
Packages (88) acl-2.2.53-1  arch-install-scripts-19-1  archlinux-keyring-20181003-1  argon2-20171227-3  attr-2.4.48-1  audit-2.8.4-2  bash-4.4.023-1  bzip2-1.0.6-8  ca-certificates-20180821-1  ca-certificates-cacert-20140824-4 [removal]  ca-certificates-mozilla-3.39-1  ca-certificates-utils-20180821-1  coreutils-8.30-1  cryptsetup-2.0.4-1  curl-7.61.1-3  db-5.3.28-4  dbus-1.12.10-2  device-mapper-2.02.181-1  e2fsprogs-1.44.4-1  expat-2.2.6-1  filesystem-2018.8-1  gawk-4.2.1-1  gcc-libs-8.2.1+20180831-1  gdbm-1.18-1  glib2-2.58.1-1  glibc-2.28-4  gnupg-2.2.10-1  gnutls-3.5.19-2  gpgme-1.11.1-2  hwids-20180518-1  iana-etc-20180913-1  icu-62.1-1  iptables-1:1.6.2-3  json-c-0.13.1-2  keyutils-1.5.11-1  kmod-25-1  krb5-1.16.1-1  libarchive-3.3.3-1  libassuan-2.5.1-1  libcap-ng-0.7.9-1  libelf-0.174-1  libgcrypt-1.8.3-1  libgpg-error-1.32-1  libidn-1.35-1  libidn2-2.0.5-1  libksba-1.3.5-1  libldap-2.4.46-2  libnftnl-1.1.1-1  libnghttp2-1.33.0-1  libnl-3.4.0-1  libpcap-1.9.0-1  libpsl-0.20.2-1  libsasl-2.1.26-13  libseccomp-2.3.3-1  libsecret-0.18.6-1  libssh2-1.8.0-2  libsystemd-239.2-1  libtasn1-4.13-1  libtirpc-1.1.4-1  libunistring-0.9.10-1  libusb-1.0.22-1  libutil-linux-2.32.1-2  linux-api-headers-4.17.11-1  lz4-1:1.8.3-1  mpfr-4.0.1-1  ncurses-6.1-3  nettle-3.4-1  npth-1.6-1  openssl-1.1.1-1  p11-kit-0.23.14-1  pacman-5.1.1-1  pacman-mirrorlist-20180912-1  pam-1.3.1-1  pambase-20171006-1  pcre-8.42-1  pcre2-10.32-1  perl-5.28.0-1  pinentry-1.1.0-4  popt-1.16-9  readline-7.0.005-1  shadow-4.6-1  sqlite-3.25.2-1  systemd-239.2-1  tzdata-2018e-2  util-linux-2.32.1-2  xz-5.2.4-1  zlib-1:1.2.11-3  zstd-1.3.5-1
Total Installed Size:  436.98 MiB
Net Upgrade Size:      111.83 MiB
:: Proceed with installation? [Y/n]
checking keyring...
downloading required keys...
Errors occurred, no packages were upgraded.
 pci WARNING:  Generated script signal 1 near or at line number 69 by !
TermuxArch pci  v1.6.5.id2161: DONE 🏁

**7. Last of all installations from the script without manual changes, except mirror selection**
> startarch s[u] user cmd: Login as user and execute command.  Use startarch addauser user first to create this user and user's home directory.
startarch  v1.6.5.id2161: DONE 🏁
proot warning: can't chdir("/data/sdext2/data/com.termux/files/home/arch/./.") in the guest rootfs: No such file or directory
proot info: default working directory is now "/"
bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
Welcome to Arch Linux in Termux!
Install a package: pacman -S package
More  information: pacman -[D|F|Q|R|S|T|U]h
Search   packages: pacman -Ss query
Upgrade  packages: pacman -Syu
Chat: wiki.termux.com/wiki/Community
Help: info query and man query
IRC:  wiki.archlinux.org/index.php/IRC_channel
bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
[proot19:03/]$

**resolv.conf is not automatically set by the script, causing an upgrade problem, but after it is set manually, it still does not fix the proot problem.**

**Comment by gauloics** [comment]
Great, this is closer to what you call a one-tap installation method when compared to the old script but unfortunately there is a problem with the proot.

**Comment by sdrausty** [comment]
@gauloics this https://github.com/sdrausty/TermuxArch/commit/fc1c6a82e79ced9f089f8ddfc3a61672b49c2d88 commit changes a couple of things regarding configuration based on your https://github.com/sdrausty/TermuxArch/issues/128#issuecomment-428286678 comments.  Hopefully, we are at one tap install for ASUS with these latest modifications.

**Comment by gauloics** [comment]
proot problem still occur

> 🕛 > 🕘 Cleaning up installation files: \e]2; 🕛 > 🕙 Cleaning up installation files: \007DONE
 🕛 > 🕤 Arch Linux in Termux PRoot is installed.  Configuring and updating Arch Linux 📲\e]2; 🕛 > 🕤 Arch Linux is installed!  Configuring and updating Arch Linux 📲 \007
**proot warning: can't chdir("/data/sdext2/data/com.termux/files/home/arch/./.") in the guest rootfs: No such file or directory
proot info: default working directory is now "/"**
:: Removing redundant packages for Termux PRoot installation…

**Comment by sdrausty** [comment]
I ran across this type of error yesterday once:
> proot warning: can't chdir("/data/sdext2/data/com.termux/files/home/arch/./.") in the guest rootfs: No such file or directory

Unfortunately, I don't remember how to reproduce it.  I will try to generate it later on today.  Hopefully we can get TermuxArch to perform as expected on ASUS [like it does now with Amazon Fire](https://github.com/sdrausty/TermuxArch/issues/131#issuecomment-428566217).  I was hoping we would achieve one tap install goal on ASUS with the last commit.  What about the errors in https://github.com/sdrausty/TermuxArch/issues/128#issuecomment-428286678?  Are they gone?

**Comment by sdrausty** [comment]
Have you tried modifying the proot statement? Comment https://github.com/sdrausty/TermuxArch/issues/132#issuecomment-428649065 has a short guide how, and these files  should be well commented, too.

@gauloics after running `$ ./setupTermuxArch.sh re`, what is the output of `[root13:03arch]$ f "exec proot" startarch` on the device?  Does it look like this?
```
exec proot --kill-on-exit --link2symlink -S /data/data/com.termux/files/home/arch -b /proc/self/fd/1:/dev/stdout -b /proc/self/fd/2:/dev/stderr /usr/bin/env -i HOME=/root TERM=xterm-256color  /bin/bash -l ||:
exec proot --kill-on-exit --link2symlink -S /data/data/com.termux/files/home/arch -b /proc/self/fd/1:/dev/stdout -b /proc/self/fd/2:/dev/stderr /usr/bin/env -i HOME=/root TERM=xterm-256color  /bin/bash -lc "$ar2ar" ||:
exec proot --kill-on-exit -S /data/data/com.termux/files/home/arch -b /proc/self/fd/1:/dev/stdout -b /proc/self/fd/2:/dev/stderr /usr/bin/env -i HOME=/root TERM=xterm-256color  /bin/su - "$ar2ar" ||:
exec proot --kill-on-exit --link2symlink -S /data/data/com.termux/files/home/arch -b /proc/self/fd/1:/dev/stdout -b /proc/self/fd/2:/dev/stderr /usr/bin/env -i HOME=/root TERM=xterm-256color  /bin/"$ar2ar" ||:
exec proot --kill-on-exit -S /data/data/com.termux/files/home/arch -b /proc/self/fd/1:/dev/stdout -b /proc/self/fd/2:/dev/stderr /usr/bin/env -i HOME=/root TERM=xterm-256color  /bin/su - "$2" -c "$ar3ar" ||:
```
The part after `-S`, is this path correct on your device?

**Comment by gauloics** [comment]
> Have you tried modifying the proot statement? Comment #132 (comment) has a short guide how, and these files should be well commented, too.
@gauloics after running $ ./setupTermuxArch.sh re, what is the output of [root13:03arch]$ f "exec proot" startarch on the device? Does it look like this?

Not sure about this, `bash setupTermuxArch.sh re` error `setupTermuxArch.sh re v1.6.5.id4119: [Exit Signal 198] DONE` for now.

> What about the errors in #128 (comment)? Are they all gone?

Not all goes, but I think, considering the source file image is downloaded from archlinux and not from archlinux32, the hard part of making it one tap is that the archlinux32 repository is different from archlinux, so it must change manually. Can your script extract an ISO image file? If it can be simpler if you download directly from archlinux32 for i686 images.

This step by step error:

**1. Proot error**
> Unpacking archlinux-bootstrap-2017.03.01-i686.tar.gz will take a long time; Be patient…
 🕛 > 🕘 Cleaning up installation files: \e]2; 🕛 > 🕙 Cleaning up installation files: \007DONE
 🕛 > 🕤 Arch Linux in Termux PRoot is installed.  Configuring and updating Arch Linux 📲\e]2; 🕛 > 🕤 Arch Linux is installed!  Configuring and updating Arch Linux 📲 \007
**proot warning: can't chdir("/data/sdext2/data/com.termux/files/home/arch/./.") in the guest rootfs: No such file or directory
proot info: default working directory is now "/"**
:: Removing redundant packages for Termux PRoot installation…

**2. Archlinux32-keyring**
*This error occurs because the repository is not from archlinux32*
> ==> Updating trust database...
gpg: marginals needed: 3  completes needed: 1  trust model: pgp
gpg: depth: 0  valid:   1  signed:   6  trust: 0-, 0q, 0n, 0m, 0f, 1u
gpg: depth: 1  valid:   6  signed:  64  trust: 0-, 0q, 0n, 6m, 0f, 0u
gpg: depth: 2  valid:  64  signed:   6  trust: 64-, 0q, 0n, 0m, 0f, 0u
gpg: next trustdb check due at 2018-12-31
==> Running pacman -Syy archlinux-keyring archlinux32-keyring ca-certificates-utils --noconfirm --color=always…
:: Synchronizing package databases...
 core                             135.4 KiB   146K/s 00:01 [################################] 100%
 extra                           1636.9 KiB   477K/s 00:03 [################################] 100%
 community                          4.6 MiB   533K/s 00:09 [################################] 100%
**error: target not found: archlinux32-keyring**
When Appending keys from archlinux.gpg appears on the screen, the installation process can be accelerated.  The system desires a lot of entropy at this part of the install procedure.  To generate as much entropy as possible quickly, watch and listen to a file on your device.

**3. When pci runing**
*still download the x86_64 architecture by default if the pacman.conf architecture value is auto, you must change to Architecture = i686*
> Total Download Size:    97.12 MiB
Total Installed Size:  437.94 MiB
Net Upgrade Size:      112.80 MiB
:: Proceed with installation? [Y/n]
:: Retrieving packages...
 linux-api-headers-4.17.11-1-any  927.7 KiB   285K/s 00:03 [################################] 100%
 tzdata-2018e-2-**x86_64**            359.0 KiB   215K/s 00:02 [################################] 100%
 iana-etc-20180913-1-any          364.9 KiB   181K/s 00:02 [################################] 100%
 filesystem-2018.8-1-**x86_64**         7.5 KiB   251K/s 00:00 [################################] 100%
 glibc-2.28-4-**x86_64**                9.1 MiB   371K/s 00:25 [################################] 100%
 gcc-libs-8.2.1+20180831-1-x...    20.3 MiB   619K/s 00:34 [################################] 100%
 ncurses-6.1-3-x86_64            1074.0 KiB   321K/s 00:03 [################################] 100%
 readline-7.0.005-1-x86_64        294.5 KiB   196K/s 00:02 [################################] 100%
**kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]]  46%** (***←not sure about this***)
 bash-4.4.023-1-x86_64           1428.3 KiB   306K/s 00:05 [################################] 100%
 zlib-1:1.2.11-3-x86_64            81.3 KiB   113K/s 00:01 [################################] 100%
...
...
pacman-5.1.1-1-x86_64            748.1 KiB   322K/s 00:02 [################################] 100%
 libtirpc-1.1.4-1-x86_64          177.5 KiB   154K/s 00:01 [################################] 100%
**kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]**
 pambase-20171006-1-any             2.9 KiB  0.00B/s 00:00 [################################] 100%
 pam-1.3.1-1-x86_64               552.8 KiB   208K/s 00:03 [################################] 100%
 shadow-4.6-1-x86_64             1097.6 KiB   303K/s 00:04 [################################] 100%
...
...

**4. Still part of number 3 process**
> systemd-239.2-1-x86_64             4.2 MiB   517K/s 00:08 [################################] 100% arch-install-scripts-19-1-any      8.9 KiB  24.1K/s 00:00 [################################] 100%
(90/90) checking keys in keyring                           [################################] 100%
downloading required keys...
error: key "EEEEE2EEEE2EEEEE" could not be looked up remotely
error: key "4A1AFC345EBE18F8" could not be looked up remotely                                     error: required key missing from keyring                                                          error: failed to commit transaction (unexpected error)                                            Errors occurred, no packages were upgraded.                                                       error: no operation specified (use -h for help)                                                                                                                                                      pci WARNING:  Generated script signal 1 near or at line number 82 by !                                                                                                                                                                                                                               TermuxArch pci gzip sed v1.6.5.id4119: DONE 🏁                                                                                                                                                      error: key "EEEEE2EEEE2EEEEE" could not be looked up remotely                                     error: key "4A1AFC345EBE18F8" could not be looked up remotely
error: required key missing from keyring
error: failed to commit transaction (unexpected error)
error: no operation specified (use -h for help)                                                                                                   finishsetup.sh WARNING:  Generated script signal 201 near or at line number 53 by
==> Running TermuxArch pci  v1.6.5.id4119 …                                                       
:: Synchronizing package databases...

**5. Last**
> util-linux-2.32.1-2  xz-5.2.4-1  zlib-1:1.2.11-3  zstd-1.3.5-1
Total Installed Size:  436.98 MiB
Net Upgrade Size:      111.83 MiB                                                                                                 :: Proceed with installation? [Y/n]
checking keyring...                                                                               downloading required keys...
Errors occurred, no packages were upgraded.                                                                                                 **pci WARNING:  Generated script signal 1 near or at line number 71 by !                                                                                                                                                                                                   TermuxArch pci  v1.6.5.id4119: DONE 🏁 **                                                                                                                                                           !
finishsetup.sh  v1.6.5.id4119 [Exit Signal 201]: DONE  🏁                                                                                                  🕛 > 🕙 Releasing wake-lock: DONE                                                                
 🕛 > 🕥 Use startarch to launch Arch Linux in Termux PRoot.  Alternatively, run ~/arch/startarch in a BASH shell to start Arch Linux in Termux PRoot for future sessions.  See startarch help for usage information.                                                                                                                                                                                   \e]2;  Thank you for using `setupTermuxArch.sh` to install Arch Linux in Termux 📲  \007 🕛 > 🕦 startarch copied to /data/data/com.termux/files/usr/bin.

*also resolv.conf still not automatically set nameserver...*

**Comment by sdrausty** [comment]
> what is the output of [root13:03arch]$ f "exec proot" startarch on the device? 

What is the output of `grep "exec proot" arch/startarch` on the device?

**Comment by gauloics** [comment]
Welcome to Termux!

Wiki:            https://wiki.termux.com
Community forum: https://termux.com/community
IRC channel:     #termux on freenode
Gitter chat:     https://gitter.im/termux/termux
Mailing list:    termux+subscribe@groups.io

Search packages:   pkg search <query>
Install a package: pkg install <package>
Upgrade packages:  pkg upgrade
Learn more:        pkg help
Hit:1 https://termux.net stable InRelease
Reading package lists... Done
Reading package lists... Done
Building dependency tree
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
—
——
————
—————» START ARCHLINUX? «—————
[Y/n]:... n
————
——
—
Ok! You can run 'startarch' at any time you want to start TermuxArch...
**$ `grep "exec proot" arch/startarch`
> exec proot --kill-on-exit --link2symlink -S /data/data/com.termux/files/home/arch -b /data/data/com.termux/files/home/arch/tmp:/dev/shm -b /dev/ -b /proc/ -b /sys/ /usr/bin/env -i HOME=/root TERM=xterm-256color  /bin/bash -l ||:
exec proot --kill-on-exit --link2symlink -S /data/data/com.termux/files/home/arch -b /data/data/com.termux/files/home/arch/tmp:/dev/shm -b /dev/ -b /proc/ -b /sys/ /usr/bin/env -i HOME=/root TERM=xterm-256color  /bin/bash -lc "$ar2ar" ||:
exec proot --kill-on-exit -S /data/data/com.termux/files/home/arch -b /data/data/com.termux/files/home/arch/tmp:/dev/shm -b /dev/ -b /proc/ -b /sys/ /usr/bin/env -i HOME=/root TERM=xterm-256color  /bin/su - "$ar2ar" ||:
exec proot --kill-on-exit --link2symlink -S /data/data/com.termux/files/home/arch -b /data/data/com.termux/files/home/arch/tmp:/dev/shm -b /dev/ -b /proc/ -b /sys/ /usr/bin/env -i HOME=/root TERM=xterm-256color  /bin/"$ar2ar" ||:
exec proot --kill-on-exit -S /data/data/com.termux/files/home/arch -b /data/data/com.termux/files/home/arch/tmp:/dev/shm -b /dev/ -b /proc/ -b /sys/ /usr/bin/env -i HOME=/root TERM=xterm-256color  /bin/su - "$2" -c "$ar3ar" ||:

$**

**Comment by gauloics** [comment]
> startarch  v1.6.7.id8209: DONE 🏁

This version successfully installed the proot, but still has to go through step by step as the top guide of this issue

**Comment by gauloics** [comment]
`locale-gen`
>Generating locales...
  en_US.UTF-8...failed to set locale!
[error] character map file `UTF-8' not found: No such file or directory
failed to set locale!
[error] default character map file `ANSI_X3.4-1968' not found: No such file or directory

**Comment by gauloics** [comment]
>locale-gen
Generating locales...
en_US.UTF-8...failed to set locale!
[error] character map file UTF-8' not found: No such file or directory failed to set locale! [error] default character map fileANSI_X3.4-1968' not found: No such file or directory

oh sorry forget to take *.pacnew file.

**Comment by gauloics** [comment]
got problem systemd
https://github.com/sdrausty/TermuxArch/issues/134

**Comment by sdrausty** [comment]
> got problem systemd

Are you also using https://gitter.im/termux to find help?

**Comment by sdrausty** [comment]
Reference https://github.com/innFactory/react-native-quick-sample/pull/1 for labels.
